.github/workflows/edr-ci.yml

```diff
@@ -84,43 +71,24 @@ jobs:
       #   uses: taiki-e/install-action@nextest
 
       # - name: Test with latest nextest release
-      #   uses: actions-rs/cargo@v1
       #   env:
       #     CARGO_INCREMENTAL: ${{ matrix.CARGO_INCREMENTAL }}
       #     ALCHEMY_URL: ${{ secrets.ALCHEMY_URL }}
-      #   with:
-      #     command: nextest
-      #     args: run --workspace --all-features --all-targets
+      #   run: cargo nextest run --workspace --all-features --all-targets
 
   edr-style:
     name: Check EDR Style
     runs-on: ubuntu-latest
     needs: check-edr
     steps:
       - uses: actions/checkout@v3
+      - uses: ./.github/actions/setup-rust
 
       - name: Install nightly rustfmt
-        uses: actions-rs/toolchain@v1
-        with:
-          toolchain: nightly
-          profile: minimal
-          components: rustfmt
-
-      - name: Install stable toolchain
-        uses: actions-rs/toolchain@v1
-        with:
-          profile: minimal
-          override: true
-          components: clippy
-
-      - uses: Swatinem/rust-cache@v2
+        run: rustup toolchain install nightly --profile minimal --component rustfmt
```

Comment:
Why not use inputs to the `setup-rust` action?

https://github.com/NomicFoundation/edr/blob/feat/multichain/.github/actions/setup-node/action.yml#L4

Reply:
I found it to be a lot simpler - we just execute `setup-rust` as our baseline "setup Rust toolchain as expected" and this is the only exception to this. Since it's a one-liner and used only here, I found it simpler to just call this directly and keep the main action lean.

Reply:
You can have default values for all inputs of an action, which would keep the default action lean too, while allowing configuration.

I'll leave it up to you to decide whether to take that approach or not.

Reply:
I tried to find a middle ground in https://github.com/NomicFoundation/edr/pull/678/commits/e7796f73696342cf6ef8fc91d73bf832ac1ff4df, hope that's okay with you.

Reply:
Thanks! That's exactly what I had in mind :)